lib/mix/tasks/exfmt.ex

```diff
@@ -6,85 +6,96 @@ defmodule Mix.Tasks.Exfmt do
 
   ## Command line options
 
+    * `--stdin` - Read from STDIN instead of a file
     * `--unsafe` - Disable the semantics check that verifies
       that `exmft` has not altered the semantic meaning of
       the input file.
 
   """
 
+  defmodule Action do
+    defstruct [:stdout, :stderr, exit_code: 0]
+  end
+
   @shortdoc  "Format Elixir source code"
   @usage """
   USAGE:
       mix exfmt path/to/file.ex
   """
 
   use Mix.Task
-  alias Exfmt.{SyntaxError, SemanticsError}
 
   @doc false
   @spec run(OptionParser.argv) :: any
   def run([]) do
-    @usage
-    |> red()
-    |> IO.write()
+    Mix.Shell.IO.error(@usage)
   end
 
   def run(args) do
-    option_parser_options = [strict: [unsafe: :boolean]]
-    with {opts, [path], []} <- OptionParser.parse(args, option_parser_options),
-         {:file, _, {:ok, source}} <- {:file, path, File.read(path)},
-         {:ok, formatted} <- format(source, opts) do
-      IO.write formatted
-    else
-      {:file, path, {:error, :enoent}} ->
-        "Error: No such file or directory:\n    #{path}"
-        |> red()
-        |> IO.puts
-
-      {:file, path, {:error, :eisdir}} ->
-        "Error: Input is a directory, not an Elixir source file:\n   #{path}"
-        |> red()
-        |> IO.puts
-
-
-      {:file, path, {:error, :eacces}} ->
-        "Error: Incorrect permissions, unable to read file:\n   #{path}"
-        |> red()
-        |> IO.puts
-
-      {:file, path, {:error, :enomem}} ->
-        "Error: Not enough memory to read file:\n   #{path}"
-        |> red()
-        |> IO.puts
-
-      {:file, path, {:error, :enotdir}} ->
-        "Error: Unable to open a parent directory:\n   #{path}"
-        |> red()
-        |> IO.puts
+    args
+    |> process
+    |> execute
+  end
 
-      %SemanticsError{message: message} ->
-        message
-        |> red()
-        |> IO.puts
+  def process(args) do
+    args
+    |> parse
+    |> input
+    |> format
+    |> output
+  end
```

Comment:
Could we move everything except `output/1` into a new module called `Exfmt.Cli` please? :)

```elixir
args
|> Cli.run()
|> output()
```